service/frontend/admin/handler_test.go

```diff
@@ -1619,3 +1621,1120 @@ func TestGetReplicationMessages(t *testing.T) {
+			},
+			dcHandlerFunc: func(mock *cache.MockDomainCache) {
+				mock.EXPECT().GetDomain("test-domain").Return(cache.NewGlobalDomainCacheEntryForTest(
+					&persistence.DomainInfo{ID: "test-domain-id"},
+					&persistence.DomainConfig{},
+					&persistence.DomainReplicationConfig{},
+					0,
+				), nil)
+			},
+			wantErr: false,
+		},
+		"refresh workflow tasks error": {
+			input: &types.RefreshWorkflowTasksRequest{
+				Domain: "test-domain",
+				Execution: &types.WorkflowExecution{
+					WorkflowID: "test-workflow-id",
+				},
+			},
+			hcHandlerFunc: func(mock *history.MockClient) {
+				mock.EXPECT().RefreshWorkflowTasks(gomock.Any(), gomock.Any()).Return(assert.AnError)
+			},
+			dcHandlerFunc: func(mock *cache.MockDomainCache) {
+				mock.EXPECT().GetDomain("test-domain").Return(cache.NewGlobalDomainCacheEntryForTest(
+					&persistence.DomainInfo{ID: "test-domain-id"},
+					&persistence.DomainConfig{},
+					&persistence.DomainReplicationConfig{},
+					0,
+				), nil)
+			},
+			wantErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run("", func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			defer ctrl.Finish()
+
+			hcMock := history.NewMockClient(ctrl)
+			dcMock := cache.NewMockDomainCache(ctrl)
+			if tt.hcHandlerFunc != nil {
+				tt.hcHandlerFunc(hcMock)
+			}
+			if tt.dcHandlerFunc != nil {
+				tt.dcHandlerFunc(dcMock)
+			}
+
+			handler := adminHandlerImpl{
+				Resource: &resource.Test{
+					Logger:        testlogger.New(t),
+					MetricsClient: metrics.NewNoopMetricsClient(),
+					HistoryClient: hcMock,
+					DomainCache:   dcMock,
+				},
+			}
+
+			err := handler.RefreshWorkflowTasks(context.Background(), tt.input)
+			if tt.wantErr {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+}
+
+func Test_ResendReplicationTasks(t *testing.T) {
+	tests := map[string]struct {
+		input         *types.ResendReplicationTasksRequest
+		hcHandlerFunc func(mock *history.MockClient)
+		wantErr       bool
+	}{
+		"nil request": {
+			input: nil,
+			hcHandlerFunc: func(mock *history.MockClient) {
+			},
+			wantErr: true,
+		},
+		"normal request": {
+			input: &types.ResendReplicationTasksRequest{
+				RemoteCluster: "test-cluster",
+			},
+			hcHandlerFunc: func(mock *history.MockClient) {
+			},
+			wantErr: true,
+		},
+	}
+
+	for name, td := range tests {
+		t.Run(name, func(t *testing.T) {
+			goMock := gomock.NewController(t)
+			hcMock := history.NewMockClient(goMock)
+			if td.hcHandlerFunc != nil {
+				td.hcHandlerFunc(hcMock)
+			}
+			handler := adminHandlerImpl{
+				Resource: &resource.Test{
+					Logger:        testlogger.New(t),
+					MetricsClient: metrics.NewNoopMetricsClient(),
+					HistoryClient: hcMock,
+				},
+			}
+
+			err := handler.ResendReplicationTasks(context.Background(), td.input)
+			if td.wantErr {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+}
+
+func Test_GetCrossClusterTasks(t *testing.T) {
```

Comment:
I think @davidporter-id-au will cleanup cross cluster related code but doesn't hurt to keep these new tests until that happens